core/src/main/java/com/amplifyframework/core/model/query/predicate/NotEqualQueryOperator.java

```diff
@@ -40,6 +40,16 @@ public Object value() {
         return value;
     }
 
+    /**
+     * Returns true if the the provided field value does
+     * not equal the value associated with this operator.
+     * @param field the field value to operate on
+     * @return evaluated result of the operator
+     */
+    public boolean evaluate(Object field) {
+        return !field.equals(value);
```

Comment:
This would rely on the fact that `field` 's class implements `equals` method. What happens when the class doesn't implement the equals method?

Reply:
`.equals` is implemented by every instance of `Object`
https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html

Reply:
Josha Block [offers this guidance on when to override `equals()`](http://thefinestartist.com/effective-java/08) in his [Effective Java 2E](https://www.amazon.com/Effective-Java-2nd-Joshua-Bloch/dp/0321356683):
```
Appropriate time to override equals method:
  - a class has a notion of logical equality that differs from mere object identity
  - a superclass has not already overridden equals to implement the desired behavior
```

A common source of programming errors in Java are classes which do not correctly implement `equals()` and `hashCode()`. In fact, we had [one such problem in the CLI code-gen](https://github.com/aws-amplify/amplify-cli/commit/5b74dfa0b0badc9a6f4d8eb8f1bbdc03e3819ecc), before re:Invent.

TLDR: we should assume that any `Object` passed into this method has a sensibly implemented `equals()`.